### PR TITLE
[SeaTunnel #967][seatunnel-connector] Add neo4j spark source 

### PR DESCRIPTION
**Pull request by an-shi-chi-fan**
## Purpose of this pull request

add neo4j source

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/configuration/source-plugins/neo4j.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# Source plugin: Neo4j
+
```
Comment (wuchunfu) [inline docs/en/configuration/source-plugins/neo4j.md:2]: Please change the title `Source plugin: Neo4j` to `Source plugin: Neo4j [Spark]`

Comment (an-shi-chi-fan) [reply docs/en/configuration/source-plugins/neo4j.md]: it was fixed

**Comment by wuchunfu** [comment]
Note that this neo4j seems to be an open source agreement of GNU. Will this conflict with the Apache open source agreement? If there is no conflict, should a declaration file be added? @CalvinKirs

**Comment by CalvinKirs** [comment]
> Note that this neo4j seems to be an open source agreement of GNU. Will this conflict with the Apache open source agreement? If there is no conflict, should a declaration file be added? @CalvinKirs

you are right, which is not compatible with AL2.

**Comment by CalvinKirs** [comment]
Sorry, I will close this PR first. Thanks again for your contribution.

**Comment by an-shi-chi-fan** [comment]
i had check it before i am writing code , this jar  was on Apache-2.0 License.

**Comment by CalvinKirs** [comment]
sorry, I read the wrong repository.
This is my rough check result
Apache XBean :: ASM 6 shaded (repackaged)

From: 'OW2' (http://www.ow2.org/)
  - asm (http://asm.ow2.org/) org.ow2.asm:asm:jar:6.2.1
    License: BSD  (http://asm.ow2.org/license.html)
  - asm-analysis (http://asm.ow2.org/) org.ow2.asm:asm-analysis:jar:6.2.1
    License: BSD  (http://asm.ow2.org/license.html)
  - asm-commons (http://asm.ow2.org/) org.ow2.asm:asm-commons:jar:6.2.1
    License: BSD  (http://asm.ow2.org/license.html)
  - asm-tree (http://asm.ow2.org/) org.ow2.asm:asm-tree:jar:6.2.1
    License: BSD  (http://asm.ow2.org/license.html)

From: 'The Apache Software Foundation' (http://www.apache.org)
  - Apache XBean :: ASM Util (http://geronimo.apache.org/maven/xbean/4.10/xbean-asm-util) org.apache.xbean:xbean-asm-util:bundle:4.10
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)

File: seatunnel-connectors/seatunnel-connector-spark-neo4j/pom.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <version>2.0.5-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>seatunnel-connector-spark-neo4j</artifactId>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+    </properties>
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connector-spark-neo4j/pom.xml:35]: this should be delete

File: seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    val notConfigMustAllParam: Array[String] = mustConfiguredAll.filter(item => !config.hasPath(item))
+    val notConfigMustOneParam: Array[String] = mustConfiguredOne.filter(item => !config.hasPath(item))
+    val isConfiguredOne: Boolean = notConfigMustOneParam.size < mustConfiguredOne.size
+
+    if (notConfigMustAllParam.isEmpty && isConfiguredOne) {
+      new CheckResult(true, "neo4j config is enough")
+    } else {
+      new CheckResult(false, s"neo4j config is not enough please check config [${notConfigMustAllParam.mkString(" ")}] " +
+        s"${if (!isConfiguredOne) "you must have one of " + mustConfiguredOne.mkString(" ") else ""}")
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala:56]: we have a public `CheckResult` Class

Comment (an-shi-chi-fan) [reply seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala:56]: this is already public `CheckResult` class, just message is too long, i will change it's style

Comment (CalvinKirs) [reply seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala]: CheckConfigUtil.check is new check method.

Comment (an-shi-chi-fan) [reply seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala]: i have seen it, this tool is need check all of the param, but i have some param thar user must config one not all. so i do not use it.  @CalvinKirs

Comment (an-shi-chi-fan) [reply seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala]:
```
 val mustConfiguredAll: Array[String] = Array("result_table_name", "url")
  val mustConfiguredOne: Array[String] = Array("query", "labels", "relationship")
```
query、 labels、relationship， user config one of it is enough

**Comment by an-shi-chi-fan** [comment]
@wuchunfu @CalvinKirs CC

File: docs/en/configuration/source-plugins/neo4j.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+```bash
+   Neo4j {
+      result_table_name = "test"
+      authentication_type = "basic"
+      authentication_basic_username = "test"
+      authentication_basic_password = "test"
+      url = "bolt://localhost:7687"
+      labels = "Person"
+      #query = "MATCH (n1)-[r]->(n2) RETURN r, n1, n2 "
+   }
```
Comment (wuchunfu) [inline docs/en/configuration/source-plugins/neo4j.md:118]: This example does not match the required options described in `Options` above

Comment (an-shi-chi-fan) [reply docs/en/configuration/source-plugins/neo4j.md:118]: it was match yes* means you can choose one of it, not all yes* must configured

File: seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    val checkALL: CheckResult = CheckConfigUtil.checkOne(config, "query", "labels", "relationship")
+    val checkOne: CheckResult = CheckConfigUtil.check(config, "result_table_name", "url")
+    CheckConfigUtil.mergeCheckMessage(checkALL, checkOne);
+
+  }
+
+  override def prepare(prepareEnv: SparkEnvironment): Unit = {
+    config.entrySet().asScala.foreach(entry => {
+      neo4jConf.put(entry.getKey.replaceAll(raw"_", raw"."), config.getString(entry.getKey))
```
Comment (RickyHuo) [inline seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala:54]: we can use '.' directly, such as `schema.flatten.limit`

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      read.option(it._1, it._2)
+    })
+    read.load()
+
+  }
+
+  override def checkConfig(): CheckResult = {
+
+    val checkALL: CheckResult = CheckConfigUtil.checkOne(config, "query", "labels", "relationship")
+    val checkOne: CheckResult = CheckConfigUtil.check(config, "result_table_name", "url")
```
Comment (RickyHuo) [inline seatunnel-connectors/seatunnel-connector-spark-neo4j/src/main/scala/org/apache/seatunnel/spark/source/Neo4j.scala]: The variable name is confused.

File: docs/en/spark/configuration/source-plugins/neo4j.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+Cypher query to read the data. you must specify  one way of (query labels relationship)
+
+### labels [string]
+
+List of node labels separated by : The first label will be the primary label. you must specify  one way of (query labels relationship)
+
+### relationship [string]
+
+Name of a relationship. you must specify  one way of (query labels relationship)
```
Comment (RickyHuo) [inline docs/en/spark/configuration/source-plugins/neo4j.md]: You must specify one option from [query, labels OR relationship]

Just a suggestion :)

**Review by RickyHuo — CHANGES_REQUESTED** [review]

**Review by RickyHuo — APPROVED** [review]
LGTM

**Pull request merged**